src/praisonai-agents/tests/unit/test_run_on_docker.py

```
@@ -0,0 +1,182 @@
+# ── the real thing ───────────────────────────────────────────────────────────
+@needs_docker
+def test_the_whole_agent_really_runs_in_a_container():
+    """Proves the loop moved: the agent reports the container's platform, not
+    the host's. Needs a model key, so it asserts placement rather than output."""
+    from praisonai.integrations.docker_managed_agent import DockerManagedAgent
+
+    backend = DockerManagedAgent(config={"instructions": "be brief"})
+    try:
+        backend._ensure_container()
+        assert backend._container, "no container was started"
+        probe = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)